runtime/interpreter/value.go

```diff
@@ -14038,6 +14058,14 @@ func (v *CompositeValue) Destroy(interpreter *Interpreter, locationRange Locatio
 		}()
 	}
 
+	// if this type has attachments, destroy all of them before invoking the destructor
+	v.forEachAttachment(interpreter, locationRange, func(attachment *CompositeValue) {
+		// an attachment's destructor may make reference to `base`, so we must set the base value
+		// for the attachment before invoking its destructor
+		attachment.setBaseValue(interpreter, v)
```

Comment:
Why do we need to explicitly set the base here? Wouldn't the destructor work similarly to any other function defined on the attachment?

Reply:
Any other function defined on the attachment must be accessed via an access expression `v[A]` on the composite value `v`. This access expression is what populates the `base` field on `A`. Destroying `v` will invoke the destructor on `A` though without going through an access expression, so we need to populate the field before calling the function.

Reply:
Gotcha. This would also be good to include as a comment in the code